pkgs/top-level/all-packages.nix

```diff
@@ -32926,10 +32926,28 @@ with pkgs;
   magma = callPackage ../development/libraries/science/math/magma { };
   clmagma = callPackage ../development/libraries/science/math/clmagma { };
 
-  mathematica = callPackage ../applications/science/math/mathematica { };
-  mathematica9 = callPackage ../applications/science/math/mathematica/9.nix { };
-  mathematica10 = callPackage ../applications/science/math/mathematica/10.nix { };
-  mathematica11 = callPackage ../applications/science/math/mathematica/11.nix { };
+  mathematica = callPackage ../applications/science/math/mathematica {
+    inherit callPackage;
+  };
+
+  mathematica-cuda = callPackage ../applications/science/math/mathematica {
+    inherit callPackage;
+    cudaSupport = true;
+  };
+
+  mathematica9 = callPackage ../applications/science/math/mathematica {
+    inherit callPackage;
+    version = "9";
+  };
+
+  mathematica10 = callPackage ../applications/science/math/mathematica {
+    inherit callPackage;
```

Comment:
Isn't it the same top-level `callPackage`? Then it's propagated automatically, without the `inherit`?

Reply:
Fixed